scipy/stats/_stats_py.py

```diff
@@ -9002,19 +9005,25 @@ def combine_pvalues(pvalues, method='fisher', weights=None):
 
 def wasserstein_distance(u_values, v_values, u_weights=None, v_weights=None):
     r"""
-    Compute the first Wasserstein distance between two 1D distributions.
+    Compute the general wasserstein-1 function between two n dimensional
+    ditributions(histograms).
 
-    This distance is also known as the earth mover's distance, since it can be
-    seen as the minimum amount of "work" required to transform :math:`u` into
-    :math:`v`, where "work" is measured as the amount of distribution weight
-    that must be moved, multiplied by the distance it has to be moved.
+    The wasserstein distance, also called the Earth mover distance or the
+    optimal transport distance, is defined as a similarity metric between
+    two probability distribution. In the discrete case, the wasserstein
+    distance can be understood as the cost of an optimal transport plan
+    between two samples. A brief and intuitive introduction can be found
+    at [2]_.
 
     .. versionadded:: 1.0.0
 
     Parameters
     ----------
     u_values, v_values : array_like
-        Values observed in the (empirical) distribution.
+        could be:
+        - Values observed in the (empirical) distribution.
+        - Value of each class in the (distribution)histogram.
```

Comment:
You are right with most of your points. It is true that 2D input should be like [vec1, vec2, ...] with each vector has same length(dimensions). And I do automatically convert any 1D input into 2D so that the dimension of both input can be easily compared. 
You are totally right about the uniform distribution example, in the context of statistics, the input values are 'outcomes' and weights are 'frequencies', in machine learning context in they can also be understood as 'data points' and 'weights'. I think I can add this in the document and also a non-unique example, if it what you mean here.

Reply:
I was hoping you could elaborate in the documentation, then. Otherwise, users will have the same questions.